clients/fides-js/src/lib/cookie.ts

```diff
@@ -266,7 +267,11 @@ export const buildCookieConsentForExperiences = (
     return cookieConsent;
   }
   experience.privacy_notices.forEach((notice) => {
-    cookieConsent[notice.notice_key] = resolveConsentValue(notice, context);
+    cookieConsent[notice.notice_key] = resolveConsentValue(
```

Comment:
it feels like we are moving in an odd circle here:

1. this function `buildCookieConsentForExperiences` (called in `initialize`) tries to `resolveConsentValue` based on the privacy notice and assign it to the cookie. however, it needs to know if the notice was `previously_consented`
2. `previous_consented` itself is determined by the cookie in `assignPreviousConsentToNotices` (also called in `initialize`
3.  so if I'm understanding this right, we are assigning `previously_consented` to `notices` based on the `cookie`, then here, we are using the `previously_consented` attr we just added to set the `cookie`? that seems circuitous?